src/analysis/convex/slope.lean

```diff
@@ -217,6 +217,128 @@ lemma strict_concave_on_iff_slope_strict_anti_adjacent :
 ⟨λ h, ⟨h.1, λ x y z, h.slope_anti_adjacent⟩,
   λ h, strict_concave_on_of_slope_strict_anti_adjacent h.1 h.2⟩
 
+lemma convex_on.secant_mono_aux1 (hf : convex_on 𝕜 s f)
+  {x y z : 𝕜} (hx : x ∈ s) (hz : z ∈ s) (hxy : x < y) (hyz : y < z) :
+  f y * (z - x) ≤ (z - y) * f x + (y - x) * f z :=
+begin
+  have hxy' : 0 < y - x := by linarith,
+  have hyz' : 0 < z - y := by linarith,
+  have hxz' : 0 < z - x := by linarith,
+  rw ← le_div_iff hxz',
+  have ha : 0 ≤ (z - y) / (z - x) := by positivity,
+  have hb : 0 ≤ (y - x) / (z - x) := by positivity,
+  calc _ = _ : _
+  ... ≤ _ : hf.2 hx hz ha hb _
+  ... = _ : _,
```

Comment:
This is very hard to follow. Do you hide the expressions in the calc because they are very long? Or for another reason?

Reply:
Presumably Heather you prefer your current spelling to `(eq.trans_le _ (hf.2 hx hz ha hb)).trans_eq _` which I think is how w would usually spell this

Reply:
I was just being lazy ... have filled them in now!